java/src/main/java/org/wildfly/openssl/SSL.java

```diff
@@ -71,6 +72,8 @@ static void init() {
                             System.loadLibrary("wfssl");
                             instance = new SSLImpl();
                         } catch (Throwable e) {
+                            //Log the initial failure to provide a clear root cause.
+                            logger.log(Level.WARNING, "Failed to load wfssl native library from system path. Will attempt to load from classpath.", e);
```

Comment:
I don't think we should log at WARN, I think Debug would be sufficient - logging an exception at WARN would lead to very verbose output that users will not be able to turn off - especially as this is only informational and the fallback mechanism may still work.